### ISSUE TITLE
please add useful function!

### ISSUE THREAD

**Issue by zheka2304** [zhuowei/MCPELauncher#1049]
I think function, that sets name of exact metadata(damage) of item or block will be very useful (for me it is). Because with it you can add multiply blocks at one ID. It can help us to solve the problem with small amount of block ids.
 I really need this function, because no block ids have left (my mod is really big and i really dont want to delete blocks, here it is, you can see this mod here - https://vk.com/factorisation2_script). And with i will be able to add much more blocks.
 So could you add it, i think it will be easy for you. Write me back please.

**Comment by NeoKat** [comment]
Hi, Eugeny. I'm fan of your mod.
But we'll get closer to the subject... Zhuowei can't add this function, becouse it's impossible due to restrictions of Minecraft PE, but really, I think Zhuowei can add it, but it will waste a LOT of time.
P. S. Written in English to make Zhuowei and other english-speaking understand this text.

**Comment by PatrikXop** [comment]
Use this! Bock.defineBlock(249,"name",[
["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],//end of block 1 (sub

**Comment by zheka2304** [comment]
NeoKat, I think you didn't quite understand me. I was talking just about setting different names to blocks with different damage and one ID, like wool (white wool, black wool, ect, names are different, as metadata, but ID is one for all wool). Not about adding more block IDa.

**Comment by PatrikXop** [comment]
..ops .. but
Bock.defineBlock(249,"name",[
["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],["Texture1",0],//end of block 1 (sub0)
Bock.defineBlock(249,"name",[
["Texture2",0],["Texture2",0],["Texture2",0],["Texture2",0],["Texture2",0],["Texture2",0],//end of block 2(sub 1)!! 1,1,true); //there u go, 2 vlocks with same aiD, but difrent damage

**Comment by NeoKat** [comment]
zheka2304, Oops... Excuse me. I'm really not surely understanded you. That can be added with some lesser time waste =3

**Comment by zheka2304** [comment]
PatrikXop, ...but with same name. It is really unpleasant detail.

**Comment by zheka2304** [comment]
I really hope that Zhuowei will add this small function :3

**Comment by PatrikXop** [comment]
Pls send me ur twitter so i can send u some screenshoots, hot to maka a difrent name, with same ID!

**Comment by PatrikXop** [comment]
And btw no chanese this function will be added, since it can be done with simple code..at least for me, since my IQ is over 167..

**Comment by zheka2304** [comment]
@zheka2304

**Comment by PatrikXop** [comment]
and look here! pls https://github.com/PatrikXop/blocks-with-same-Id-but-difrent-name/blob/master/example.js

**Comment by zheka2304** [comment]
And what about names in workbench? Or in furnace?

**Comment by PatrikXop** [comment]
well....  tick thet will check when furnance is used, and if slot = 249 and data == 0, sets item with id 8000 //(item will have name of waned ore so it would tehnicaly work..) and for chest when player useItem on chest it will check all items in chest, if ther it is it will replace them with id 8000 //if sub is 0 or 8001 //if sub is 1 and so on, and another mod tick thet will check if player ever hase item 8000 in hande, if has it will replace it with starting ore........it could be done, but it will take areound 900 lines.. and it might be a bot laggy

**Comment by PatrikXop** [comment]
be a bit laggy*

**Comment by PatrikXop** [comment]
But for workbench .. i am not shure

**Comment by zheka2304** [comment]
Yeah, it will be really laggy and too big for such small task. And, you might know, Zhuowei added function to set custom shape to specified block metadata. So if he will see it, he will maybe add this for names.

**Comment by zheka2304** [comment]
So we can just wait for it.

**Comment by Aternix** [comment]
Why you don't try to make a new function in your script?
